File: VB/4PosBackOffice.NET/frmIncrementQuantity.vb
Option Strict Off
Option Explicit On
Friend Class frmIncrementQuantity
	Inherits System.Windows.Forms.Form
	Dim gIncrementID As Integer
	Dim gQuantity As Integer
	
	Private Sub loadLanguage()
		
		'frmIncrementQuantity = No Code [Edit Increment Quantity]
		'rsLang.filter = "LanguageLayoutLnk_LanguageID=" & 0000
		'If rsLang.RecordCount Then frmIncrementQuantity.Caption = rsLang("LanguageLayoutLnk_Description"): frmIncrementQuantity.RightToLeft = rsLang("LanguageLayoutLnk_RightTL")
		
		rsLang.filter = "LanguageLayoutLnk_LanguageID=" & 1004 'Exit|Checked
		If rsLang.RecordCount Then cmdClose.Text = rsLang.Fields("LanguageLayoutLnk_Description").Value : cmdClose.RightToLeft = rsLang.Fields("LanguageLayoutLnk_RightTL").Value
		
		'_lbl_1 = No Code               [Increment Name]
		'rsLang.filter = "LanguageLayoutLnk_LanguageID=" & 0000
		'If rsLang.RecordCount Then _lbl_1.Caption = rsLang("LanguageLayoutLnk_Description"): _lbl_1.RightToLeft = rsLang("LanguageLayoutLnk_RightTL")
		
		rsLang.filter = "LanguageLayoutLnk_LanguageID=" & 2145 'Shrink Size|Checked
		If rsLang.RecordCount Then _lblLabels_9.Text = rsLang.Fields("LanguageLayoutLnk_Description").Value : _lblLabels_9.RightToLeft = rsLang.Fields("LanguageLayoutLnk_RightTL").Value
		
		rsLang.filter = "LanguageLayoutLnk_LanguageID=" & 1151 'Price|Checked
		If rsLang.RecordCount Then _lbl_3.Text = rsLang.Fields("LanguageLayoutLnk_Description").Value : _lbl_3.RightToLeft = rsLang.Fields("LanguageLayoutLnk_RightTL").Value
		
		rsHelp.filter = "Help_Section=0 AND Help_Form='" & Me.Name & "'"
		'UPGRADE_ISSUE: Form property frmIncrementQuantity.ToolTip1 was not upgraded. Click for more: 'ms-help://MS.VSCC.v90/dv_commoner/local/redirect.htm?keyword="CC4C7EC0-C903-48FC-ACCC-81861D12DA4A"'
		If rsHelp.RecordCount Then Me.ToolTip1 = rsHelp.Fields("Help_ContextID").Value
		
	End Sub
	
	Private Sub loadData()
		Dim rs As ADODB.Recordset
		If gQuantity Then
			rs = getRS("SELECT IncrementQuantity.*, Increment.Increment_Name FROM Increment INNER JOIN IncrementQuantity ON Increment.IncrementID = IncrementQuantity.IncrementQuantity_IncrementID WHERE (((IncrementQuantity.IncrementQuantity_IncrementID)=" & gIncrementID & ") AND ((IncrementQuantity.IncrementQuantity_Quantity)=" & gQuantity & "));")
			If rs.RecordCount Then
				lblStockItem.Text = rs.Fields("Increment_Name").Value
				txtPrice.Text = CStr(rs.Fields("IncrementQuantity_Price").Value * 100)
				txtPrice_Leave(txtPrice, New System.EventArgs())
				txtQuantity.Text = CStr(gQuantity)
				txtQuantity_Leave(txtQuantity, New System.EventArgs())
			Else
				Me.Close()
				Exit Sub
			End If
		Else
			rs = getRS("SELECT Increment.* From Increment WHERE (((Increment.IncrementID)=" & gIncrementID & "));")
			If rs.RecordCount Then
				lblStockItem.Text = rs.Fields("Increment_Name").Value
				txtPrice.Text = CStr(0)
				txtPrice_Leave(txtPrice, New System.EventArgs())
				txtQuantity.Text = CStr(gQuantity)
				txtQuantity_Leave(txtQuantity, New System.EventArgs())
			Else
				Me.Close()
				Exit Sub
			End If
		End If
	End Sub
	Public Sub loadItem(ByRef incrementID As Integer, ByRef quantity As Integer)
		gIncrementID = incrementID
		gQuantity = quantity
		loadData()
		
		loadLanguage()
		ShowDialog()
	End Sub
	Private Sub frmIncrementQuantity_KeyPress(ByVal eventSender As System.Object, ByVal eventArgs As System.Windows.Forms.KeyPressEventArgs) Handles MyBase.KeyPress
		Dim KeyAscii As Short = Asc(eventArgs.KeyChar)
        Dim mbAddNewFlag As Boolean
        Dim mbEditFlag As Boolean
		If mbEditFlag Or mbAddNewFlag Then GoTo EventExitSub
		
		Select Case KeyAscii
			Case System.Windows.Forms.Keys.Escape
				KeyAscii = 0
				cmdClose.Focus()
				System.Windows.Forms.Application.DoEvents()
				cmdClose_Click(cmdClose, New System.EventArgs())
		End Select
EventExitSub: 
		eventArgs.KeyChar = Chr(KeyAscii)
		If KeyAscii = 0 Then
			eventArgs.Handled = True
		End If
	End Sub
	
	Private Sub cmdCancel_Click()
        Dim mbDataChanged As Boolean
        Dim mvBookMark As Integer
        Dim adoPrimaryRS As ADODB.Recordset
        Dim mbAddNewFlag As Boolean
        Dim mbEditFlag As Boolean
		On Error Resume Next
		If mbAddNewFlag Then
			Me.Close()
		Else
			mbEditFlag = False
			mbAddNewFlag = False
			adoPrimaryRS.CancelUpdate()
			If mvBookMark > 0 Then
				adoPrimaryRS.Bookmark = mvBookMark
			Else
				adoPrimaryRS.MoveFirst()
			End If
			mbDataChanged = False
		End If
	End Sub
	
	Private Function update_Renamed() As Boolean
		On Error GoTo UpdateErr
		Dim lBit As Short
		update_Renamed = True
		Me.cmdClose.Focus()
		System.Windows.Forms.Application.DoEvents()
		cnnDB.Execute("DELETE IncrementQuantity.* From IncrementQuantity WHERE (((IncrementQuantity.IncrementQuantity_IncrementID)=" & gIncrementID & ") AND ((IncrementQuantity.IncrementQuantity_Quantity)=" & gQuantity & "));")
		cnnDB.Execute("DELETE IncrementQuantity.* From IncrementQuantity WHERE (((IncrementQuantity.IncrementQuantity_IncrementID)=" & gIncrementID & ") AND ((IncrementQuantity.IncrementQuantity_Quantity)=" & txtQuantity.Text & "));")
		If CShort(txtQuantity.Text) > 1 Then
			cnnDB.Execute("INSERT INTO IncrementQuantity ( IncrementQuantity_IncrementID, IncrementQuantity_Quantity, IncrementQuantity_Price ) SELECT " & gIncrementID & ", " & txtQuantity.Text & ", " & txtPrice.Text & ";")
		End If
		Exit Function
UpdateErr: 
		MsgBox(Err.Description)
		update_Renamed = True
	End Function
	
	Private Sub cmdClose_Click(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles cmdClose.Click
		cmdClose.Focus()
		System.Windows.Forms.Application.DoEvents()
		If update_Renamed() Then
			Me.Close()
		End If
	End Sub
	Private Sub txtPrice_Enter(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles txtPrice.Enter
        MyGotFocusNumeric(txtPrice)
	End Sub
	
	Private Sub txtPrice_KeyPress(ByVal eventSender As System.Object, ByVal eventArgs As System.Windows.Forms.KeyPressEventArgs) Handles txtPrice.KeyPress
		Dim KeyAscii As Short = Asc(eventArgs.KeyChar)
        MyKeyPressNegative(txtPrice, KeyAscii)
		eventArgs.KeyChar = Chr(KeyAscii)
		If KeyAscii = 0 Then
			eventArgs.Handled = True
		End If
	End Sub
	
	Private Sub txtPrice_Leave(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles txtPrice.Leave
        MyLostFocus(txtPrice, 2)
	End Sub
	
	Private Sub txtQuantity_Enter(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles txtQuantity.Enter
        MyGotFocusNumeric(txtQuantity)
	End Sub
	
	Private Sub txtQuantity_KeyPress(ByVal eventSender As System.Object, ByVal eventArgs As System.Windows.Forms.KeyPressEventArgs) Handles txtQuantity.KeyPress
		Dim KeyAscii As Short = Asc(eventArgs.KeyChar)
        MyKeyPressNegative(txtQuantity, KeyAscii)
		eventArgs.KeyChar = Chr(KeyAscii)
		If KeyAscii = 0 Then
			eventArgs.Handled = True
		End If
	End Sub
	
	Private Sub txtQuantity_Leave(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles txtQuantity.Leave
        MyLostFocus(txtQuantity, 0)
	End Sub
End Class

File: C-Sharp/frmIncrementQuantity.cs
using CrystalDecisions.CrystalReports.Engine;
using CrystalDecisions.Windows.Forms;
using DpSdkEngLib;
using DPSDKOPSLib;
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Drawing;
using System.Diagnostics;
using System.Windows.Forms;
using System.Linq;
using System.Xml.Linq;
 // ERROR: Not supported in C#: OptionDeclaration
namespace _4PosBackOffice.NET
{
	internal partial class frmIncrementQuantity : System.Windows.Forms.Form
	{
		int gIncrementID;
		int gQuantity;

		private void loadLanguage()
		{

			//frmIncrementQuantity = No Code [Edit Increment Quantity]
			//rsLang.filter = "LanguageLayoutLnk_LanguageID=" & 0000
			//If rsLang.RecordCount Then frmIncrementQuantity.Caption = rsLang("LanguageLayoutLnk_Description"): frmIncrementQuantity.RightToLeft = rsLang("LanguageLayoutLnk_RightTL")

			modRecordSet.rsLang.filter = "LanguageLayoutLnk_LanguageID=" + 1004;
			//Exit|Checked
			if (modRecordSet.rsLang.RecordCount){cmdClose.Text = modRecordSet.rsLang.Fields("LanguageLayoutLnk_Description").Value;cmdClose.RightToLeft = modRecordSet.rsLang.Fields("LanguageLayoutLnk_RightTL").Value;}

			//_lbl_1 = No Code               [Increment Name]
			//rsLang.filter = "LanguageLayoutLnk_LanguageID=" & 0000
			//If rsLang.RecordCount Then _lbl_1.Caption = rsLang("LanguageLayoutLnk_Description"): _lbl_1.RightToLeft = rsLang("LanguageLayoutLnk_RightTL")

			modRecordSet.rsLang.filter = "LanguageLayoutLnk_LanguageID=" + 2145;
			//Shrink Size|Checked
			if (modRecordSet.rsLang.RecordCount){_lblLabels_9.Text = modRecordSet.rsLang.Fields("LanguageLayoutLnk_Description").Value;_lblLabels_9.RightToLeft = modRecordSet.rsLang.Fields("LanguageLayoutLnk_RightTL").Value;}

			modRecordSet.rsLang.filter = "LanguageLayoutLnk_LanguageID=" + 1151;
			//Price|Checked
			if (modRecordSet.rsLang.RecordCount){_LBL_3.Text = modRecordSet.rsLang.Fields("LanguageLayoutLnk_Description").Value;_LBL_3.RightToLeft = modRecordSet.rsLang.Fields("LanguageLayoutLnk_RightTL").Value;}

			modRecordSet.rsHelp.filter = "Help_Section=0 AND Help_Form='" + this.Name + "'";
			//UPGRADE_ISSUE: Form property frmIncrementQuantity.ToolTip1 was not upgraded. Click for more: 'ms-help://MS.VSCC.v90/dv_commoner/local/redirect.htm?keyword="CC4C7EC0-C903-48FC-ACCC-81861D12DA4A"'
			if (modRecordSet.rsHelp.RecordCount)
				this.ToolTip1 = modRecordSet.rsHelp.Fields("Help_ContextID").Value;

		}

		private void loadData()
		{
			ADODB.Recordset rs = default(ADODB.Recordset);
			if (gQuantity) {
				rs = modRecordSet.getRS(ref "SELECT IncrementQuantity.*, Increment.Increment_Name FROM Increment INNER JOIN IncrementQuantity ON Increment.IncrementID = IncrementQuantity.IncrementQuantity_IncrementID WHERE (((IncrementQuantity.IncrementQuantity_IncrementID)=" + gIncrementID + ") AND ((IncrementQuantity.IncrementQuantity_Quantity)=" + gQuantity + "));");
				if (rs.RecordCount) {
					lblStockItem.Text = rs.Fields("Increment_Name").Value;
					txtPrice.Text = Convert.ToString(rs.Fields("IncrementQuantity_Price").Value * 100);
					txtPrice_Leave(txtPrice, new System.EventArgs());
					txtQuantity.Text = Convert.ToString(gQuantity);
					txtQuantity_Leave(txtQuantity, new System.EventArgs());
				} else {
					this.Close();
					return;
				}
			} else {
				rs = modRecordSet.getRS(ref "SELECT Increment.* From Increment WHERE (((Increment.IncrementID)=" + gIncrementID + "));");
				if (rs.RecordCount) {
					lblStockItem.Text = rs.Fields("Increment_Name").Value;
					txtPrice.Text = Convert.ToString(0);
					txtPrice_Leave(txtPrice, new System.EventArgs());
					txtQuantity.Text = Convert.ToString(gQuantity);
					txtQuantity_Leave(txtQuantity, new System.EventArgs());
				} else {
					this.Close();
					return;
				}
			}
		}
		public void loadItem(ref int incrementID, ref int quantity)
		{
			gIncrementID = incrementID;
			gQuantity = quantity;
			loadData();

			loadLanguage();
			ShowDialog();
		}
		private void frmIncrementQuantity_KeyPress(System.Object eventSender, System.Windows.Forms.KeyPressEventArgs eventArgs)
		{
			short KeyAscii = Strings.Asc(eventArgs.KeyChar);
			bool mbAddNewFlag = false;
			bool mbEditFlag = false;
			if (mbEditFlag | mbAddNewFlag)
				goto EventExitSub;

			switch (KeyAscii) {
				case System.Windows.Forms.Keys.Escape:
					KeyAscii = 0;
					cmdClose.Focus();
					System.Windows.Forms.Application.DoEvents();
					cmdClose_Click(cmdClose, new System.EventArgs());
					break;
			}
			EventExitSub:
			eventArgs.KeyChar = Strings.Chr(KeyAscii);
			if (KeyAscii == 0) {
				eventArgs.Handled = true;
			}
		}

		private void cmdCancel_Click()
		{
			bool mbDataChanged = false;
			int mvBookMark = 0;
			ADODB.Recordset adoPrimaryRS = default(ADODB.Recordset);
			bool mbAddNewFlag = false;
			bool mbEditFlag = false;
			 // ERROR: Not supported in C#: OnErrorStatement

			if (mbAddNewFlag) {
				this.Close();
			} else {
				mbEditFlag = false;
				mbAddNewFlag = false;
				adoPrimaryRS.CancelUpdate();
				if (mvBookMark > 0) {
					adoPrimaryRS.Bookmark = mvBookMark;
				} else {
					adoPrimaryRS.MoveFirst();
				}
				mbDataChanged = false;
			}
		}

		private bool update_Renamed()
		{
			bool functionReturnValue = false;
			 // ERROR: Not supported in C#: OnErrorStatement

			short lBit = 0;
			functionReturnValue = true;
			this.cmdClose.Focus();
			System.Windows.Forms.Application.DoEvents();
			modRecordSet.cnnDB.Execute("DELETE IncrementQuantity.* From IncrementQuantity WHERE (((IncrementQuantity.IncrementQuantity_IncrementID)=" + gIncrementID + ") AND ((IncrementQuantity.IncrementQuantity_Quantity)=" + gQuantity + "));");
			modRecordSet.cnnDB.Execute("DELETE IncrementQuantity.* From IncrementQuantity WHERE (((IncrementQuantity.IncrementQuantity_IncrementID)=" + gIncrementID + ") AND ((IncrementQuantity.IncrementQuantity_Quantity)=" + txtQuantity.Text + "));");
			if (Convert.ToInt16(txtQuantity.Text) > 1) {
				modRecordSet.cnnDB.Execute("INSERT INTO IncrementQuantity ( IncrementQuantity_IncrementID, IncrementQuantity_Quantity, IncrementQuantity_Price ) SELECT " + gIncrementID + ", " + txtQuantity.Text + ", " + txtPrice.Text + ";");
			}
			return functionReturnValue;
			UpdateErr:
			Interaction.MsgBox(Err().Description);
			functionReturnValue = true;
			return functionReturnValue;
		}

		private void cmdClose_Click(System.Object eventSender, System.EventArgs eventArgs)
		{
			cmdClose.Focus();
			System.Windows.Forms.Application.DoEvents();
			if (update_Renamed()) {
				this.Close();
			}
		}
		private void txtPrice_Enter(System.Object eventSender, System.EventArgs eventArgs)
		{
			modUtilities.MyGotFocusNumeric(ref txtPrice);
		}

		private void txtPrice_KeyPress(System.Object eventSender, System.Windows.Forms.KeyPressEventArgs eventArgs)
		{
			short KeyAscii = Strings.Asc(eventArgs.KeyChar);
			modUtilities.MyKeyPressNegative(ref txtPrice, ref KeyAscii);
			eventArgs.KeyChar = Strings.Chr(KeyAscii);
			if (KeyAscii == 0) {
				eventArgs.Handled = true;
			}
		}

		private void txtPrice_Leave(System.Object eventSender, System.EventArgs eventArgs)
		{
			modUtilities.MyLostFocus(ref txtPrice, ref 2);
		}

		private void txtQuantity_Enter(System.Object eventSender, System.EventArgs eventArgs)
		{
			modUtilities.MyGotFocusNumeric(ref txtQuantity);
		}

		private void txtQuantity_KeyPress(System.Object eventSender, System.Windows.Forms.KeyPressEventArgs eventArgs)
		{
			short KeyAscii = Strings.Asc(eventArgs.KeyChar);
			modUtilities.MyKeyPressNegative(ref txtQuantity, ref KeyAscii);
			eventArgs.KeyChar = Strings.Chr(KeyAscii);
			if (KeyAscii == 0) {
				eventArgs.Handled = true;
			}
		}

		private void txtQuantity_Leave(System.Object eventSender, System.EventArgs eventArgs)
		{
			modUtilities.MyLostFocus(ref txtQuantity, ref 0);
		}
	}
}
